ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/transaction/filtered/UtxoFilteredTransaction.java

```diff
@@ -124,4 +126,7 @@ public interface UtxoFilteredTransaction extends TransactionWithMetadata {
      * @throws CordaRuntimeException if the current {@link UtxoFilteredTransaction} fails to verify correctly.
      */
     void verify();
+
+    @NotNull
+    Map<Integer, MerkleProof> getMerkleProofs();
```

Comment:
Would it be better to make this an operation which takes a set of integers as n argument and returns exactly the `MerkleProof` for each requested integer? If callers ever don't need certain component groups we could be wasting a noticeable effort in many cases.